docs/DesignDocs/CryptoLibConfiguration.md

```diff
@@ -0,0 +1,115 @@
+MBEDTLS_PKCS1_V15 | N/A | - |
+MBEDTLS_PKCS1_V21 | N/A | - |
+!MBEDTLS_RSA_NO_CRT | N/A | Cannot be disabled on OpenSSL. |
+MBEDTLS_AESNI_C | AES_ASM | - |
+MBEDTLS_AES_C | N/A | - |
+!MBEDTLS_ARC4_C | OPENSSL_NO_RC4 | Disable on OpenSSL with the no-rc4 configuration. |
+MBEDTLS_BIGNUM_C | N/A | - |
+!MBEDTLS_BLOWFISH_C | OPENSSL_NO_BF | Drop support for uncommon cipher. Disable on OpenSSL with the no-bf configuration. |
+!MBEDTLS_CAMELLIA_C | OPENSSL_NO_CAMELLIA | Drop support for uncommon cipher. Disable on OpenSSL with the no-camellia configuration. |
+!MBEDTLS_ARIA_C | OPENSSL_NO_ARIA | Drop support for uncommon cipher. Disable on OpenSSL with the no-aria configuration. |
+MBEDTLS_CCM_C | N/A | - |
+!MBEDTLS_CHACHA20_C | OPENSSL_NO_CHACHA | Disable less common cipher until there's a demand for it. Disable on OpenSSL with the no-chacha configuration. |
+!MBEDTLS_CHACHAPOLY_C | OPENSSL_NO_CHACHA | Disable less common cipher until there's a demand for it. Disable on OpenSSL with the no-chacha configuration. |
+MBEDTLS_CMAC_C | !OPENSSL_NO_CMAC | Enable as it's broadly used, allowed by NIST SP standards. |
+MBEDTLS_DES_C | !OPENSSL_NO_DES | Enable for backward compatibility as some protocols use it (e.g. payment industry protocols). |
+MBEDTLS_ECDH_C | !OPENSSL_NO_EC | - |
+MBEDTLS_ECDSA_C | !OPENSSL_NO_EC | - |
+MBEDTLS_ECJPAKE_C | N/A | OpenSSL does not support JPAKE (removed in v1.1.0). |
+MBEDTLS_ECP_C | N/A | - |
+MBEDTLS_GCM_C | N/A | - |
+!MBEDTLS_HAVEGE_C | N/A | Mbed-specific option. |
+MBEDTLS_HKDF_C | N/A | OpenSSL supports HKDF by default (started from v1.1.0). |
+MBEDTLS_HMAC_DRBG_C | N/A | Mbed-specific option. |
+!MBEDTLS_NIST_KW_C | N/A | No equivalent option in OpenSSL. |
+MBEDTLS_MD_C | N/A | No equivalent option in OpenSSL. |
+!MBEDTLS_MD2_C | OPENSSL_NO_MD2 | Disable on OpenSSL with the no-md2 configuration. |
+!MBEDTLS_MD4_C | OPENSSL_NO_MD4 | Disable on OpenSSL with the no-md4 configuration. |
+MBEDTLS_MD5_C | !OPENSSL_NO_MD5 | Enable for backward compatibility. The algorithm is still commonly used. |
+MBEDTLS_PKCS5_C | N/A | - |
+!MBEDTLS_PKCS11_C | N/A | - |
+MBEDTLS_PKCS12_C | N/A | - |
+!MBEDTLS_POLY1305_C | OPENSSL_NO_POLY1305 | Drop the uncommon hash algorithm to minimize TCB. Disable on OpenSSL with the no-poly1305 configuration.
+!MBEDTLS_RIPEMD160_C | OPENSSL_NO_RMD160 | Drop the uncommon hash algorithm to minimize TCB. Disable on OpenSSL with the no-rmd160 configuration. |
+MBEDTLS_RSA_C | !OPENSSL_NO_RSA | - |
+MBEDTLS_SHA1_C | N/A | OpenSSL supports SHA1 by default. |
+MBEDTLS_SHA256_C | N/A | OpenSSL supports SHA256 by default. |
+MBEDTLS_SHA512_C | N/A | OpenSSL supports SHA512 by default. |
+!MBEDTLS_XTEA_C | N/A | No equivalent option in OpenSSL. |
+
+## OpenSSL-specific Configuration
+
+OpenSSL | Comment
+|:---|:---|
+OPENSSL_NO_BLAKE2 | Blake2 hash is not supported by MbedTLS. Disable on OpenSSL with the no-blake2 configuration to minimize TCB. |
```

Comment:
```suggestion
OPENSSL_NO_BLAKE2 | Blake2 hash is not supported by Mbed TLS. Disable on OpenSSL with the no-blake2 configuration to minimize TCB. |
```
nit: In prose, there should be a space in "Mbed TLS" (see https://en.wikipedia.org/wiki/Mbed_TLS and https://github.com/ARMmbed/mbedtls).